### PR TITLE
Avoid NPE, profileFileSupplier can be null for tests

### PR DESCRIPTION
**Pull request by sugmanue**
## Motivation and Context

We have seen random failures for `UseGlobalEndpointResolverTest` and there's a red herring log when attempting to get from `profileFileSupplier` which can be `null`. This change does not fix the bug but removes the misleading warning log by avoiding dereferencing a null object. 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/UseGlobalEndpointResolver.java [hunk 1]
```diff
@@ -82,6 +82,9 @@ private static String envVarSetting() {
 
     private String profileFileSetting(Supplier<ProfileFile> profileFileSupplier, Supplier<String> profileNameSupplier) {
         try {
+            if (profileFileSupplier == null) {
+                return null;
+            }
             ProfileFile profileFile = profileFileSupplier.get();
             String profileName = profileNameSupplier.get();
```
Comment (zoewangg) [inline services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/UseGlobalEndpointResolver.java:89]: Maybe check `profileNameSupplier` as well?

Comment (sugmanue) [reply services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/UseGlobalEndpointResolver.java:89]: sure

Comment (sugmanue) [reply services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/endpoints/UseGlobalEndpointResolver.java:89]: Done.

File: response.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"statusCode": 200, "body": "\"Wheel spinned and selected zoewang\""}
```
Comment (sugmanue) [inline response.json:1]: Sorry, I will remove.

**Review by zoewangg — APPROVED** [review]

**Pull request merged**